asv/environment.py

```diff
@@ -311,7 +359,8 @@ def hashname(self):
         """
         Get a hash to uniquely identify this environment.
         """
-        return hashlib.md5(self.name.encode('utf-8')).hexdigest()
+        full_name = "{0}-{1}".format(self.tool_name, self.name).encode('utf-8')
```

Comment:
Would it make sense to just have tool_name included as part of the name property -- since tool_name is now required to uniquely identify the environment...